subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopySpecIntegrationSpec.groovy

```diff
@@ -194,4 +196,28 @@ class CopySpecIntegrationSpec extends AbstractIntegrationSpec {
         file('dest/😀.txt').exists()
         false // TODO This test can pass on Windows with proper locale, this force the test to fail, remove once fixed
     }
+
+    @Requires(TestPrecondition.UNIX_DERIVATIVE)
+    @Issue("https://github.com/gradle/gradle/issues/2552")
+    def "can copy files to output with named pipes"() {
+        def input = file("input.txt").createFile()
+
+        def outputDirectory = file("output").createDir()
+        def pipe = outputDirectory.file("testPipe").createNamedPipe()
+
+        buildFile << """
+            task copy(type: Copy) {
+                from '${input.name}'
+                into '${outputDirectory.name}'
+            }
+        """
+
+        when:
+        run "copy"
+        then:
+        outputDirectory.list().contains input.name
+
+        cleanup:
+        pipe.delete()
```

Comment:
The TestNameTestDirectoryProvider already deletes all the files in the temporary directory. Why do you need to add this extra cleanup?